src/filter/mod.rs

```diff
@@ -11,13 +11,127 @@ use image::{GenericImage, GenericImageView, GrayImage, ImageBuffer, Luma, Pixel,
 use crate::definitions::{Clamp, Image};
 use crate::integral_image::{column_running_sum, row_running_sum};
 use crate::map::{ChannelMap, WithChannel};
-use num::Num;
+use num::{abs, pow, Num};
 
 use crate::math::cast;
 use conv::ValueInto;
 use std::cmp::{max, min};
 use std::f32;
 
+/// Bilateral filtering of grayscale images.
```

Comment:
It would be nice to have a link to an explanation of bilateral filtering, and/or a doc test. But neither is a blocker to merging.